ci: Add documentation check workflow

evelikov opened the pull request:
We really don't build the docs in CI. Due to this a number of issues have crept in.

---
There is a bit of a caveat here - as-is the CI fails, although I don't have enough background knowledge to resolve the issues. Glancing around, some of the issues were introduced by the current maintainers - so I hope they can provide fixes.

jangsc0000 committed:
docs: add encoder interface for av1

v2 (Emil)
 - update expectations

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
ci: Add documentation check workflow

We really don't build the docs in CI. Due to this a number of issues
have crept in.

v2:
 - compare against in-tree expectations

v3:
 - drop paths-ignore hunk

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

dvrogozh committed:
docs: fix doxygen markup

v2 (Emil)
 - update expectations

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

dvrogozh committed:
docs: fix references and descriptions

v2 (Emil)
 - update expectations

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

dvrogozh reviewed (changes requested):
Well, we are not ready to check for warnings/errors from doxygen at the moment. There is really a bunch of problems and  we will need some time to fix.

So, let's break this into stages. I hope #661 will be the first one to add "fatal" docs build ci and addressing few high level trivial problems.

evelikov commented:
A slightly better idea came to mind, inspired by Mesa:
 - store the expectation in-tree
 - fail the pipeline if the output changes - fixes need to update the expectations, new changes must not add errors

Will update the MR, to include your fixes as an illustration.

evelikov committed:
meson: add dummy doxygen wrapper doxywrap.sh

Seems that ninja will fold stderr into stdout, making that output a bit
garbled. But more importantly, it's impossible for us to track the
actual issues reported by doxygen.

We want to store those as baseline expectations, this way we can fix
them as time goes by and also flag/prevent any new ones creeping in.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov commented:
If the documentation is going through refactor (like the last patch), the approach seems a little noisy. But overall something everyone can live with in the short to mid term.

evelikov commented:
@dvrogozh what is your view on the updated approach?

dvrogozh reviewed (changes requested):
> @dvrogozh what is your view on the updated approach?

We don't have that many errors from doxygen actually to invent the regression checker. I think we need to attempt fix them. As such I suggest to go step by step and first merge in #661 which adds minor fixes + adds a build path check.

evelikov commented:
> Overall I agree that fixing the issues would be the better approach. Then again it goes into the "perfect is the enemy of good" category.
> 
> Regardless, say that we fix have a fix for all of the issues - how can we ensure, in automated way, that new issues don't get introduced?
> 
> The proposed regression tracker trivially handles that, while also helping us to get there. Doesn't it?

@dvrogozh does the above make sense?

evelikov commented:
@dvrogozh @XinfengZhang humble poke - is there anything I can to do move this forward?

evelikov commented:
Ping?

XinfengZhang commented:
looks there are different opinions , could we merge #681  as first step, then continue the discussion of  the next step ? we could decouple them ...
